api/src/main/java/org/pmiops/workbench/notebooks/LeonardoNotebooksClientImpl.java

```diff
@@ -97,8 +107,18 @@ private LeonardoCreateRuntimeRequest buildCreateRuntimeRequest(
     nbExtensions.put(
         "aou-upload-policy-extension", assetsBaseUrl + "/aou-upload-policy-extension.js");
 
+    Map<String, String> runtimeLabels = new HashMap<>();
+    runtimeLabels.put(RUNTIME_LABEL_AOU, "true");
+    runtimeLabels.put(RUNTIME_LABEL_CREATED_BY, userEmail);
+
+    if (runtime.getConfigurationType() != null) {
+      runtimeLabels.put(
+          RUNTIME_LABEL_AOU_CONFIG,
+          RUNTIME_CONFIGURATION_TYPE_ENUM_TO_STORAGE_MAP.get(runtime.getConfigurationType()));
+    }
+
     return new LeonardoCreateRuntimeRequest()
-        .labels(ImmutableMap.of(RUNTIME_LABEL_AOU, "true", RUNTIME_LABEL_CREATED_BY, userEmail))
+        .labels(runtimeLabels)
```

Comment:
The machineType and masterDisk size are still not being passed through on L135-140. The logic should be as follows:

- If Runtime is provided, prefer the masterMachineType and masterDiskSize from the client-specified values.
- Else if !enableCustomRuntimes && clusterOverride != null, prefer that (with current fallback logic for either type or disk)
- else use the default from config.firecloud.*

Optionally: require a runtime payload be provided if enableCustomRuntimes; potentially this mitigates the need for bullet (2) above

Reply:
I thought we said that there isn't too much value in shuttling the default values from the API to the UI just for it to come back. Given that the upcoming UI work will be overwriting that bit of code anyways.

I'll make the changes but I really don't think either version will block or hamper the upcoming tickets in the sprint, ones I cannot work on ATM until this merges.